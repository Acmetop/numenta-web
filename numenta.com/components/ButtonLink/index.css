._disabled:disabled {
  background-color: #999 !important;
}

.primary {
  composes: bn from 'tachyons';
  composes: f4 from 'tachyons';
  composes: f5-ns from 'tachyons';
  composes: ph4 from 'tachyons';
  composes: pv3 from 'tachyons';
  composes: white from 'tachyons';
  composes: bg-color-blue from '../../static/assets/css/theme.css';
}

.buttonLink {
  composes: br2 from 'tachyons';
  composes: dib from 'tachyons';
  composes: dim from 'tachyons';
  composes: flex-auto from 'tachyons';
  composes: link from 'tachyons';
  composes: ma0 from 'tachyons';
  composes: pointer from 'tachyons';
  composes: shadow-4 from 'tachyons';
  composes: _disabled;
}
