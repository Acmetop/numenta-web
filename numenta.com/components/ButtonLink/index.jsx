// Numenta company website source code. Copyright Â© 2016 Numenta.
// Full details in LICENSE.txt, or contact us at <http://numenta.com>.
// This program is free software: you can redistribute it and/or modify it under
// the terms of the GNU Affero General Public License as published by the Free
// Software Foundation, either version 3 of the License, or (at your option) any
// later version. This program is distributed in the hope that it will be
// useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
// General Public License for more details. You should have received a copy of
// the GNU Affero General Public License along with this program. If not, see
// <https://www.gnu.org/licenses/agpl.html>.

import classNames from 'classnames'
import React from 'react'

import styles from './index.css'


/**
 * External link only (browser location) for now.
 */
const ButtonLink = ({children, disabled, onClick, theme, to}) => {
  const classes = [styles.buttonLink, styles[theme]]

  if (disabled) classes.push(styles.disabled)
  else classes.push(styles.enabled)

  return (
    <a
      className={classNames(...classes)}
      href={to}
      onClick={onClick}
    >
      {children}
    </a>
  )
}

ButtonLink.propTypes = {
  children: React.PropTypes.any.isRequired,
  disabled: React.PropTypes.bool,
  onClick: React.PropTypes.func,
  theme: React.PropTypes.string,
  to: React.PropTypes.string.isRequired,
}

ButtonLink.defaultProps = {
  disabled: false,
  theme: 'primary',
}

export default ButtonLink
