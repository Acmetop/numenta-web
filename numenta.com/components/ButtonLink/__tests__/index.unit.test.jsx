import React from 'react'
import renderer from 'react-test-renderer'

import ButtonLink from '../../ButtonLink'


describe('<ButtonLink /> React component unit test suite', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <ButtonLink to="http://numenta.com">
        Visit Numenta
      </ButtonLink>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
