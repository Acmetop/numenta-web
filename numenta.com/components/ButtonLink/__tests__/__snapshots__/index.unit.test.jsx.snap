exports[`<ButtonLink /> React component unit test suite Renders correctly 1`] = `
<a
  className="buttonLink primary enabled"
  href="http://numenta.com"
  onClick={undefined}>
  Visit Numenta
</a>
`;
